### ISSUE TITLE
Change Promise#done to take an Exit[E, A]

### ISSUE THREAD

**Issue by iravid** [zio/zio#1250]
Rationale: we have `Promise#fail(E)`, `Promise#halt(Cause[E])`, `Promise#die(Throwable)`. These match the `ZIO.fail/halt/die` constructors.

The existing `Promise#done` should be renamed to `Promise#completeWith`.

Thoughts? @ghostdogpr @jdegoes

Signatures:
```scala
class Promise[E, A] {
  def done(exit: Exit[E, A]): UIO[Boolean]
  def completeWith(io: IO[E, A]): UIO[Boolean]
}
```

**Comment by jdegoes** [comment]
💯 

Consistency matters a lot for usability.

**Comment by mgibowski** [comment]
I would be glad to make this change.

**Comment by mtsokol** [comment]
I would like to try to do it.

**Comment by ghostdogpr** [comment]
Closed by #1251
